Exercises/NoModules/Chapter 16/Soln16_05/Soln16_05.cpp

```diff
@@ -0,0 +1,125 @@
+// Exercise 16-5
+/* Somewhat artificial example to practice lists of different, 
+ * related exception types (mind the order + catch-by-reference!)
+ * and rethrowing (catch-by-reference + "throw;" !)
+ */
+
+#include <iostream>
+#include <random>     // For random number generation
+#include <functional> // For std::bind()
+#include "CurveBall.h"
+#include "DomainExceptions.h"
+
+void askEvenNumber();           // Ask the user to provide an even number
+
+int main()
+{
+  try
+  {
+    askEvenNumber();
+  }
+  catch (const CurveBall& /*caught*/)
+  {
+    std::cerr << "...hit it out of the park!" << std::endl;
+  }
+}
+
+/* Helper functions for askEvenNumber() */
+void throwCurveBallSometimes(); // Throw a CurveBall exception 25% of the time
+int readEvenNumber();           // Reads an even number from std::cin and verifies the input 
+                                // (throws upon failure)
+
+// Option 1: use recursion
+void askEvenNumber()
+{
+  try
+  {
+    std::cout << "Please enter an even number: ";
+    readEvenNumber();
+  }
+  catch (const NotANumber& nan)
+  {
+    std::cerr << nan.what() << std::endl;
+    return;
+  }
+  catch (const std::domain_error& domainException)
+  {
+    std::cerr << domainException.what() << std::endl;
+    askEvenNumber();  // Recursive call
+  }
+  catch (const std::exception& exception)
+  {
+    std::cerr << exception.what() << std::endl;
+    throw;
+  }
+}
+
+/*
+// Option 2: use a loop
+void askEvenNumber()
+{
+  bool repeat{ false };
```

Comment:
You don't really need a repeat variable right?
You could have an infinite loop and break from it.